internal/remoteconfig/remoteconfig.go

```diff
@@ -92,37 +102,65 @@ func NewClient(config ClientConfig) (*Client, error) {
 		stop:         make(chan struct{}),
 		lastError:    nil,
 		callbacks:    []Callback{},
+		capabilities: map[Capability]struct{}{},
+		products:     map[string]struct{}{},
 	}, nil
 }
 
-// Start starts the client's update poll loop in a fresh goroutine
-func (c *Client) Start() {
-	go func() {
-		ticker := time.NewTicker(c.PollInterval)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-c.stop:
-				close(c.stop)
-				return
-			case <-ticker.C:
-				c.updateState()
+// Start starts the client's update poll loop in a fresh goroutine.
+// Noop if the client has already started.
+func Start(config ClientConfig) error {
+	if client != nil {
+		return nil
+	}
+	var err error
+	client, err = newClient(config)
```

Comment:
I think the client creation + `client.startOnce()` should be a thread safe block. Otherwise we could end up with a global client with a startOnce that actually hasn't been called, even though a goroutine is running and using it.

Here is what I'm trying to describe (g1 and g2 beeing 2 separate goroutines both calling `Start()`):

g1: newClient() -------  startOnceStart -------------- startOnceEnd
g2: -------------------------------------newClient() -------------

I think a simple solution in this case would be to move the global client assignment at the end of the startOnce function. Am I making sense?

Reply:
Nice catch! I reworked the logic a bit and included the client creation in the `startOnce`.

Reply:
LGTM now